cf_remote/spawn.py

```diff
@@ -298,6 +298,69 @@ def get_cloud_driver(provider, creds, region):
     return driver
 
 
+# the string platform_name can be platform, platform-version(partial even), or platform-version-architecture
+# The data in cloud_data.py aws_image_criteria can have general information for just
+# `platform` or include all the components if necessary.
+#
+# Generally up-to-date versions should use a generic criteria which pulls the most up to date
+# image for that platform and version.
+def _get_image_criteria(platform_name):
+    log.debug("Looking for AWS AMI for platform_name '%s'" % (platform_name))
+    platform = platform_name.split("-")[0]
```

Comment:
Please add an example `platform_name` string here. Otherwise the code below is quite confusing.

Reply:
Also, looks like `platform_name` should be `.split("-")` and then the individual components should be "named" (assigned to properly named variables).